src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -1567,7 +1567,7 @@ trait Contexts { self: Analyzer =>
     private def transformImport(selectors: List[ImportSelector], sym: Symbol): List[Symbol] = selectors match {
       case List() => List()
       case List(ImportSelector(nme.WILDCARD, _, _, _)) => List(sym)
-      case ImportSelector(from, _, to, _) :: _ if from == sym.name =>
+      case ImportSelector(from, _, to, _) :: _ if from == (if (from.isTermName) sym.name.toTermName else sym.name.toTypeName) =>
```

Comment:
Does `Name` deserve `name_==` alongside weird `string_==`?

Reply:
I can't think of cases when it makes sense to equate Term/Type-names outside of import selectors, so I opted not to introduce a method on `Name` that might be misused.

Reply:
OK, next time I'll just use `string_==` 😺